randovania/server/bot.py

```diff
@@ -77,8 +716,23 @@ async def on_message(self, message: discord.Message):
         if self.configuration["channel_name_filter"] in channel.name:
             await look_for_permalinks(message.content, channel)
 
-
 def run():
+    # read state from disk if it exists
+    global async_races
+    try:
+        if os.path.exists(ASYNC_RACE_DATA_FILENAME):
+            with open(ASYNC_RACE_DATA_FILENAME, "r") as f:
+                json_string = f.read()
+                races = json.loads(json_string)
+                for race in races:
+                    async_races.append(AsyncRace.from_json(race))
+    except Exception as e:
+        exc_type, exc_obj, exc_tb = sys.exc_info()
```

Comment:
This block could use more specific Exception handling, or maybe not handling at all since corrupted data sounds very fatal.